### DISCUSSION TITLE
`.md` suffix in href of navbars

### DISCUSSION THREAD

**Discussion by mrdrivingduck** [vuepress/core#762] [Q&A] [answered]
Hi community. I'm new to VuePress 2. Recently I reference the configs of VuePress 2 official documentation website to build a website for my own project. VuePress looks great, but there is one strange thing.

I found VuePress's official website configs its navbar like:

```ts
  {
    text: 'Reference',
    children: [
      {
        text: 'VuePress',
        children: [
          {
            text: 'CLI',
            link: '/reference/cli.html',
          },
          '/reference/config.md',
          ...
```

Turns out the website is rendered as `<a href="xxx.html">`

![image](https://user-images.githubusercontent.com/32560442/159110692-36aa3ec6-78b3-4c08-8043-4ae929ad08bf.png)

![image](https://user-images.githubusercontent.com/32560442/159110703-fe3da77a-bf5e-44b4-93b2-6bf30923102b.png)

In my project, I have configured in a similar way, however the website is rendered as `<a href="xxx.md">`, so that Algolia's docs search crawler cannot work well. Although it is ok to be clicked by my mouse, it will jump to that page correctly.

```js
  {
    text: "Architecture",
    link: "/architecture/",
    children: [
      {
        text: "Overview",
        link: "/architecture/README.md",
      },
      {
        text: "Buffer Management",
        link: "/architecture/buffer-management.md",
      },
      {
        text: "DDL Synchronization",
        link: "/architecture/ddl-synchronization.md",
      },
      {
        text: "LogIndex",
        link: "/architecture/logindex.md",
      },
    ],
```

![image](https://user-images.githubusercontent.com/32560442/159111040-47e7bb4c-f842-4090-9315-2e24fd9919c6.png)

Have I missed something? My project website is [here](https://apsaradb.github.io/PolarDB-for-PostgreSQL/).

**Comment by Mister-Hope** [comment]
Hi there, not pretty much sure either, I will start to investiagte this  when I have time.(So you have to wait)

But here are some possiable reasons you may find out yourself. Open devserver, and make sure vue devtools installed. Go to vue-router page in devtool and inspect the routes vuepress registered, and compare if your link matches one of them.

I guess they may not match due to problems like UPPERCASE lowercase issues. If you find something, welcome to show and tell.

**Comment by Mister-Hope** [comment, marked as answer]
Hey, you can only use `.md` links when using them directly as strings instead of object. It's designed like that, object config is designed to keep as is.
